### PR TITLE
Remove config dependency from `OpenVINOInferencer`

### PR DESCRIPTION
**Pull request by samet-akcay**
# Description

- OpenVINO Inferencer currently expects `config` argument. Ideally, the parameters within `config` should be in `metadata`. This PR addresses this problem. 

- Fixes # (issue)

## Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Refactor (non-breaking change which refactors the code base)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist

- [ ] My code follows the [pre-commit style and check guidelines](https://github.com/openvinotoolkit/anomalib/blob/main/CONTRIBUTING.md) of this project.
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing tests pass locally with my changes
- [ ] I have added a summary of my changes to the [CHANGELOG](https://github.com/openvinotoolkit/anomalib/blob/main/CHANGELOG.md) (not for minor changes, docs and tests).

### CODE REVIEW

File: anomalib/deploy/export.py [hunk 1]
```diff
@@ -50,39 +52,77 @@ def get_model_metadata(model: AnomalyModule) -> dict[str, Tensor]:
...
+
+    return metadata
+
+
 def export(
-    model: AnomalyModule,
-    input_size: list[int] | tuple[int, int],
-    export_mode: ExportMode,
-    export_root: str | Path,
+    trainer: pl.Trainer,
```
Comment (ashwinvaidya17) [inline anomalib/deploy/export.py:83]: The only extra thing we get from trainer is the task and transform. Can't we pass this information explicitly to this method. We can reuse this for the CLI.

File: anomalib/deploy/export.py [hunk 2]
```diff
@@ -50,39 +52,77 @@ def get_model_metadata(model: AnomalyModule) -> dict[str, Tensor]:
...
+
+    metadata: dict[str, Any] = {
+        "task": trainer.datamodule.test_data.task,
+        "transform": trainer.datamodule.test_data.transform.to_dict(),
+        "image_threshold": model.image_threshold.cpu().value.item(),
+        "pixel_threshold": model.pixel_threshold.cpu().value.item(),
+    }
+
+    # Add normalization metrics to the post processing metadata
+    if hasattr(model, "normalization_metrics") and model.normalization_metrics.state_dict() is not None:
```
Comment (ashwinvaidya17) [inline anomalib/deploy/export.py:74]: What if we call `get_metadata_from_model` here?

File: CHANGELOG.md [hunk 3]
```diff
@@ -94,7 +94,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 - add deprecation warning to denormalize class by @djdameln in <https://github.com/openvinotoolkit/anomalib/pull/629>
 - Anomalib CLI Improvements - Update metrics and create post_processing section in the config file by @samet-akcay in <https://github.com/openvinotoolkit/anomalib/pull/607>
 - Convert adaptive_threshold to Enum in configs by @samet-akcay in <https://github.com/openvinotoolkit/anomalib/pull/637>
-- Create meta_data.json with ONNX export as well as OpenVINO export by @calebmm in <https://github.com/openvinotoolkit/anomalib/pull/636>
+- Create metadata.json with ONNX export as well as OpenVINO export by @calebmm in <https://github.com/openvinotoolkit/anomalib/pull/636>
```
Comment (ashwinvaidya17) [inline CHANGELOG.md:97]: Is this change needed? I feel we shouldn't be modifying history. Maybe we can add another entry stating that the key has been renamed.

**Review by ashwinvaidya17 — COMMENTED** [review]
Thanks. A few minor comments

File: requirements/base.txt [hunk 4]
```diff
@@ -4,7 +4,7 @@ einops>=0.3.2
 freia>=0.2
 imgaug==0.4.0
 jsonargparse[signatures]>=4.3
-kornia>=0.6.6
+kornia>=0.6.6,<0.6.10
```
Comment (djdameln) [inline requirements/base.txt:7]: Why is this needed?

**Review by djdameln — COMMENTED** [review]
Thanks, much better this way!
